pkg/debug/debug.go

```diff
@@ -0,0 +1,99 @@
+//  Copyright (c) 2019 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package debug
+
+import (
+	_ "expvar"
+	"log"
+	"net/http"
+	_ "net/http/pprof"
+	"os"
+	"strings"
+
+	"github.com/pkg/profile"
+)
+
+const defaultServerAddr = ":1234"
+
+var (
+	profileMode     = os.Getenv("DEBUG_PROFILE_MODE")
+	profilePath     = os.Getenv("DEBUG_PROFILE_PATH")
+	debugServerAddr = os.Getenv("DEBUG_SERVER_ADDR")
+)
+
+type Debug struct {
+	closer func()
+}
+
+func Start() interface {
+	Stop()
```

Comment:
Question: this is very interesting and I didn't know it is possible - but is there any advantage against just returning `*Debug`?

Reply:
To make it clear to the user that at that point only `Stop` should be called.